homeassistant/components/mqtt/models.py

```diff
@@ -289,12 +295,45 @@ def async_render_with_possible_json_value(
         return rendered_payload
 
 
+class EntityMonitor:
+    """Monitors entity state changes."""
+
```

Comment:
You could add `__slots__` since there is no magic here.

I know it makes it less reusable but maybe for MQTT is easier to put it in the base entity class